clients/drcachesim/tracer/tracer.cpp

```diff
@@ -104,24 +104,30 @@ DR_DISALLOW_UNSAFE_STATIC
 namespace dynamorio {
 namespace drmemtrace {
 namespace {
-#ifdef BUILD_TRACER_WITH_DR_SYSCALL
+#ifdef BUILD_DRMEMTRACE_WITH_DR_SYSCALL
 #    define SYSCALL_RECORD_BUFFER_SIZE 8192
 
 file_t syscall_record_file;
 int syscall_record_file_offset = 0;
 char syscall_record_buffer[SYSCALL_RECORD_BUFFER_SIZE];
 
+// write_syscall_record() is a per-syscall callback function. The syscall_record_buffer
+// is used to batch records and defer I/O to improve performance by reducing the the
+// frequency of file writes.
 size_t
 write_syscall_record(char *buf, size_t size)
 {
     size_t byte_written = 0;
     size_t remaining = size;
 
     while (remaining + syscall_record_file_offset >= SYSCALL_RECORD_BUFFER_SIZE) {
+        // Write partial record to fill up the buffer before writing to the file.
+        // If a memory content record's size exceeds the buffer capacity, the record is
+        // broken into multiple buffers and written one buffer at a time.
```

Comment:
If it's larger than our buffer, isn't it simpler to just write it all out at once?